server/src/main/java/com/cloud/api/query/QueryManagerImpl.java

```diff
@@ -3760,6 +3767,22 @@ protected ResourceDetailResponse createResourceDetailsResponse(ResourceDetail re
         return resourceDetailResponse;
     }
 
+    @Override
+    public ListResponse<ManagementServerResponse> listManagementServers(ListMgmtsCmd cmd) {
+        ListResponse<ManagementServerResponse> response = new ListResponse<>();
+        List<ManagementServerResponse> result = new ArrayList<>();
+        for (ManagementServerHostVO mgmt : managementServerHostDao.listAll()) {
```

Comment:
Will this `listAll` list removed entries as well (entries where the `removed` field is not null)?

Reply:
No, `listAll` does not include removed entries by default.

Reply:
Good. I am also assuming that it returns an empty list, in case it does not found any management server (unlike event).

Reply:
Another shortcut, I'm not doing any pagination since it's kind of overkill for the number of management servers.